src/main/java/soot/util/ArrayNumberer.java

```diff
@@ -52,19 +54,31 @@ private void resize(int n) {
     numberToObj = Arrays.copyOf(numberToObj, n);
   }
 
+  @Override
   public synchronized void add(E o) {
     if (o.getNumber() != 0) {
       return;
     }
 
-    ++lastNumber;
+    int chosenNumber = -1;
```

Comment:
Kommentar: "In case we removed entries from the numberer, we want to re-use the free space"